fl4health/strategies/feddg_ga_with_adaptive_constraint.py

```diff
@@ -0,0 +1,268 @@
+        evaluate_metrics_aggregation_fn: Optional[MetricsAggregationFn] = None,
+        initial_loss_weight: float = 1.0,
+        adapt_loss_weight: bool = False,
+        loss_weight_delta: float = 0.1,
+        loss_weight_patience: int = 5,
+        weighted_train_losses: bool = False,
+        fairness_metric: Optional[FairnessMetric] = None,
+        weight_step_size: float = 0.2,
+    ):
+        """
+        Strategy for the FedDG-GA algorithm (Federated Domain Generalization with Generalization Adjustment,
+        Zhang et al. 2023) combined with the Adaptive Strategy for Auxiliary constraints like FedProx. See
+        documentation on FedAvgWithAdaptiveConstraint for more information.
+
+        NOTE: Initial parameters are NOT optional. They must be passed for this strategy.
+
+        Args:
+            fraction_fit : float, optional
+                Fraction of clients used during training. In case `min_fit_clients`
+                is larger than `fraction_fit * available_clients`, `min_fit_clients`
+                will still be sampled. Defaults to 1.0.
+            fraction_evaluate : float, optional
+                Fraction of clients used during validation. In case `min_evaluate_clients`
+                is larger than `fraction_evaluate * available_clients`, `min_evaluate_clients`
+                will still be sampled. Defaults to 1.0.
+            min_fit_clients : int, optional
+                Minimum number of clients used during training. Defaults to 2.
+            min_evaluate_clients : int, optional
+                Minimum number of clients used during validation. Defaults to 2.
+            min_available_clients : int, optional
+                Minimum number of total clients in the system. Defaults to 2.
+            evaluate_fn :
+                Optional[
+                    Callable[[int, NDArrays, Dict[str, Scalar]],
+                    Optional[Tuple[float, Dict[str, Scalar]]]]
+                ]
+                Optional function used for validation. Defaults to None.
+            on_fit_config_fn : Callable[[int], Dict[str, Scalar]], optional
+                Function used to configure training. Defaults to None.
+            on_evaluate_config_fn : Callable[[int], Dict[str, Scalar]], optional
+                Function used to configure validation. Defaults to None.
+            accept_failures : bool, optional
+                Whether or not accept rounds containing failures. Defaults to True.
+            initial_parameters : Parameters, optional
+                Initial global model parameters.
+            fit_metrics_aggregation_fn : Optional[MetricsAggregationFn]
+                Metrics aggregation function, optional.
+            evaluate_metrics_aggregation_fn : Optional[MetricsAggregationFn]
+                Metrics aggregation function, optional.
+            adapt_loss_weight (bool, optional): Determines whether the value of mu is adaptively modified by
```

Comment:
nit: Should we clarify what mu is or just refer to it as loss weight? mu isn't used isnt used in the FedGdGa paper so it may be a bit confusing

Reply:
Yes, will do. It shouldn't really be called mu.